lms/djangoapps/courseware/access.py

```diff
@@ -257,6 +257,21 @@ def _can_enroll_courselike(user, courselike):
         log.warning("ENROLLMENT DENIED in %s because @ in username %s" % (course_key, user.username))
         return ACCESS_DENIED
 
+    if settings.FEATURES.get('RESTRICT_ENROLL_SOCIAL_PROVIDERS'):
+        try:
+            social_providers = user.social_auth.all()
+        except AttributeError:
+            social_providers = []
+        allowed_providers = settings.FEATURES.get('RESTRICT_ENROLL_SOCIAL_PROVIDERS')
+        if social_providers and not any(
+            [any([social_provider.uid.startswith(allowed_provider)
+                  for allowed_provider in allowed_providers])
+             for social_provider in social_providers]):
+            log.warning(
+                "ENROLLMENT DENIED in %s because user %s not authenticated by %s"
+                % (course_key, user.username, allowed_provider))
```

Comment:
You probably want to pass the args for the formatting as additional positional args to `log.warning` instead, the main reason for doing this is that sentry dedupes on the `msg` arg - if that's different for every combination of this tuple you'll get many different errors reported for the same log statement.

Reply:
:+1: thanks for the heads up, I'll adjust